File: next/dweb-browser/src/Platforms/iOS/MicroModule/NativeUI/NativeUiNMM.cs
using DwebBrowser.MicroService.Core;
using DwebBrowser.Platforms.iOS.MicroModule.NativeUI.SafeArea;
using DwebBrowser.Platforms.iOS.MicroModule.NativeUI.StatusBar;
using DwebBrowser.Platforms.iOS.MicroModule.NativeUI.NavigationBar;
using DwebBrowser.Platforms.iOS.MicroModule.NativeUI.VirtualKeyboard;

namespace DwebBrowser.MicroService.Sys.NativeUI;

public class NativeUiNMM : NativeMicroModule
{
    public NativeUiNMM() : base("nativeui.browser.dweb")
    {
    }

    private StatusBarNMM _statusBarNMM = new();
    private NavigationBarNMM _navigationBarNMM = new();
    private VirtualKeyboardNMM _virtualKeyboardNMM = new();
    private SafeAreaNMM _safeAreaNMM = new();

    protected override async Task _bootstrapAsync(IBootstrapContext bootstrapContext)
    {
        bootstrapContext.Dns.Install(_statusBarNMM);
        bootstrapContext.Dns.Install(_navigationBarNMM);
        bootstrapContext.Dns.Install(_virtualKeyboardNMM);
        bootstrapContext.Dns.Install(_safeAreaNMM);
    }
}

File: android/app/src/main/java/info/bagen/dwebbrowser/microService/browser/nativeui/NativeUiNMM.kt
package info.bagen.dwebbrowser.microService.browser.nativeui

import info.bagen.dwebbrowser.microService.browser.nativeui.navigationBar.NavigationBarNMM
import info.bagen.dwebbrowser.microService.browser.nativeui.safeArea.SafeAreaNMM
import info.bagen.dwebbrowser.microService.browser.nativeui.splashScreen.SplashScreenNMM
import info.bagen.dwebbrowser.microService.browser.nativeui.statusBar.StatusBarNMM
import info.bagen.dwebbrowser.microService.browser.nativeui.virtualKeyboard.VirtualKeyboardNMM
import org.dweb_browser.microservice.core.BootstrapContext
import org.dweb_browser.microservice.core.NativeMicroModule
import org.dweb_browser.microservice.help.MICRO_MODULE_CATEGORY

class NativeUiNMM : NativeMicroModule("nativeui.browser.dweb", "nativeUi") {

  override val categories =
    mutableListOf(MICRO_MODULE_CATEGORY.Service, MICRO_MODULE_CATEGORY.Render_Service);

  private val navigationBarNMM = NavigationBarNMM()
  private val statusBarNMM = StatusBarNMM()
  private val safeAreaNMM = SafeAreaNMM()
  private val virtualKeyboardNMM = VirtualKeyboardNMM()
  private val splashScreenNMM = SplashScreenNMM()

  override suspend fun _bootstrap(bootstrapContext: BootstrapContext) {
//        bootstrapContext.dns.install(navigationBarNMM)
//        bootstrapContext.dns.install(statusBarNMM)
//        bootstrapContext.dns.install(safeAreaNMM)
//        bootstrapContext.dns.install(virtualKeyboardNMM)
//        bootstrapContext.dns.install(splashScreenNMM)
  }

  override suspend fun _shutdown() {
  }
}